### PR TITLE
Update HCA components to USWDS v3

### PR DESCRIPTION
**Pull request by longmd**
## Summary
On February 16 2024, the Design System team will be switching all web components to render their v3 variant automatically. This PR updates the applicable components to their v3 variants and updates all unit and end-to-end testing to avoid any conflicts.

## Related issue(s)
department-of-veterans-affairs/va.gov-team#75394

## Testing done

- [x] Updated all unit tests that may have been affected
- [x] Updated all Cypress tests to fix issues with checkbox clicks

## Acceptance criteria

- All applicable components utilize their v3 variants
- v3 Design System variants do not break any of the testing

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### CODE REVIEW

**Review by JoshingYou1 — DISMISSED** [review]
Looks great to me. Awesome job, as usual, Matt! 😃

File: src/applications/hca/components/FormPages/FinancialConfirmation.jsx [hunk 1]
```diff
@@ -6,12 +6,11 @@ const DisabilityConfirmation = props => {
   const { goBack, goForward } = props;
 
   return (
-    <va-alert
+    <va-featured-content
```
Comment (micahchiang) [inline src/applications/hca/components/FormPages/FinancialConfirmation.jsx:9]: @longmd - FYSA, there's a PR out to update this component from `va-featured-content` to `va-summary-box`. - https://github.com/department-of-veterans-affairs/vets-website/pull/27945

It hasn't been merged in just yet, but it's out there. Just putting it on your radar.

**Review by micahchiang — DISMISSED** [review]

File: src/applications/hca/components/FormPages/FinancialConfirmation.jsx (hunk 1 above)
Comment (longmd) [reply src/applications/hca/components/FormPages/FinancialConfirmation.jsx:9]: I had seen that name change in Storybook yesterday. I'm curious if there is going to be a crossover where they'll both be active and we'll be told to switch over? I looked to see how many apps were using `va-featured-content` and I think it was like 6-8 at the moment.

Comment (longmd) [reply src/applications/hca/components/FormPages/FinancialConfirmation.jsx:9]: Nevermind... I see that they are being renamed in the PR. I will hold merging all three of my apps until that PR gets merged and to take advantage of the name change.

**Review by JoshingYou1 — APPROVED** [review]
LGTM! Thank you for making these QOL improvements Matt!

**Pull request merged**